packages/react-hooks/src/merkle/useMerkleGroups.ts

```diff
@@ -0,0 +1,32 @@
+import { useCall } from '@webb-dapp/react-hooks';
+import { GroupTree } from '@webb-tools/types/interfaces';
+import { useMemo } from 'react';
+import { hexToU8a } from '@polkadot/util';
+
+class MerkleGroupWrapper {
+  constructor(private _inner?: GroupTree) {}
+
+  get inner() {
+    return this._inner;
+  }
+
+  get ready() {
+    return Boolean(this.inner);
+  }
+
+  get rootHash() {
+    return this.inner?.toHuman().root_hash ?? null;
+  }
+
+  get rootHashU8a(): Uint8Array | null {
+    const rootHash = this.rootHash;
+    return rootHash ? hexToU8a(rootHash as any) : null;
+  }
+}
+
+export const useMerkleGroups = (id?: string | undefined) => {
```

Comment:
I guess the ID must be required, as we already querying for the group tree, right?
```suggestion
export const useGroupTree = (id: number): GroupTreeWrapper => {
```

Reply:
No, it's not required the code handles that case

Reply:
I agree with the renaming. If the API call is returning a GroupTree, then the hook should return that type + Wrapper.